tests/integration/test_sns.py

```diff
@@ -2330,3 +2328,172 @@ def validate_content():
             sleep_before = 10
 
         retry(validate_content, retries=retries, sleep_before=sleep_before, sleep=sleep)
+
+    @pytest.mark.only_localstack
+    def test_publish_to_platform_endpoint_can_retrospect(
+        self, sns_client, sns_create_topic, sns_subscription
+    ):
+        sns_backend = SNSBackend.get()
+        # clean up the saved messages
+        sns_backend_endpoint_arns = list(sns_backend.platform_endpoint_messages.keys())
```

Comment:
I guess this is a "something, somewhere doesn't clean up" countermeasure? :) This would of course be nicer if we could it eliminate it at the source, but I get it. It somewhat turns it into a whitebox test as a consequence, but that is probably fine for a @only_localstack test.

Reply:
I think we don't clean up by default as we store every published message, we can't really delete them. They are there purely as a retrospective way to check the messages. They have no value logic wise in the provider.